code/game/machinery/cloning.dm

```diff
@@ -165,7 +165,7 @@ GLOBAL_LIST_INIT(cloner_biomass_items, list(\
 	for(var/i=new_SE.len;i<=DNA_SE_LENGTH;i++)
 		new_SE += rand(1,1024)
 	buf.dna.SE=new_SE
-	buf.dna.SetSEValueRange(GLOB.monkeyblock,0xDAC, 0xFFF)
+	buf.dna.SetDNAValueRange(GLOB.monkeyblock, 0xDAC, 0xFFF, DNA_SE)
```

Comment:
@blinkdog 
> I also need to check cloning.dm:168 against the function signature for setting a range.

`buf.dna.SetDNAValueRange(GLOB.monkeyblock, 0xDAC, 0xFFF, DNA_SE)`
\---------------
```
/datum/dna/proc/SetDNAValueRange(block, value, maxvalue, dna_type, defer = FALSE)
var/range = (4095 / maxvalue)`
SetDNAValue(block, round(value * range) - rand(1, range - 1), DNA_SE)
```
\---------------
0xDAC --> 3500
0xFFF --> 4095

SetDNAValueRange(GLOB.monkeyblock, 3500, 4095, DNA_SE, FALSE)
\---------------
var/range = (4095 / 4095) = 1
SetDNAValue(GLOB.monkeyblock, round(3500 * 1) - rand(1, 1 - 1), DNA_SE)
\---------------
SetDNAValue(GLOB.monkeyblock, 3500 - rand(1, 0), DNA_SE)
\---------------
SetDNAValue(GLOB.monkeyblock, 3500, DNA_SE)
\---------------

SE uses Value Ranges to randomize the numbers on the DNA Modifier machines in Genetics. It's a really cool system. It encodes using hexadecimal here.

_This_, on the other hand, is hard-coding a value-range set. I'm not sure what the purpose is. This _should_ be using `SetDNAValue`.
\---------------
```
/datum/dna/proc/SetDNAValue(block, value, dna_type, defer = FALSE)`
		if(DNA_SE)
			SE[block] = value
			dirtySE = TRUE
	if(!defer)
		UpdateDNA(dna_type)		
```
\---------------
SetDNAValue(GLOB.monkeyblock, 3500, DNA_SE)

defer --> FALSE

SE[GLOB.monkeyblock] = 3500
dirtySE = TRUE

UpdateDNA(DNA_SE)
\---------------
```
/datum/dna/proc/UpdateDNA(dna_type)
		if(DNA_SE)
			struc_enzymes = ""
			for(var/block in SE)
				struc_enzymes += EncodeDNABlock(block)
			dirtySE = FALSE
```
\---------------
struc_enzymes += EncodeDNABlock(GLOB.monkeyblock)

`/proc/EncodeDNABlock(value)
	return add_zero2(num2hex(value, 1), 3)`

add_zero2(num2hex(3500, 1), 3)
add_zero2(0xDAC, 3)
0000xDAC

struc_enzymes += 0000xDAC
\---------------